### PR TITLE
fix: Added "children" to autocomplete list for multiTree and singleTree widgets

### LINKED ISSUE #23641: [Bug]: Autocomplete does not work for child objects of treeSelect and multiTreeSelect widgets
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Autocomplete suggestions do not show up for child objects of the treeSelect widgets

### Steps To Reproduce

1. Add treeSelect and multiTreeSelect widgets to canvas
2. Try to access child objects of these widgets and observe no autocomplete suggestions shows up 

image

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
Added "children" to autocomplete list for multiTree and singleTree widgets

#### PR fixes following issue(s)
Fixes #23641
>
>
#### Media
Screenshot 2023-06-05 at 12 24 10

#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
>
#### How Has This Been Tested?
- [x] Manual
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

**Review by Rishabh-Rathod — DISMISSED** [review]

**Comment by shwetha-ramesh** [comment]
@arunvjn QA Done on [DP](https://appsmith-rj4e2i4pv-get-appsmith.vercel.app/). Working as expected
image

File: app/client/cypress/e2e/Regression/ClientSide/Autocomplete/Autocomplete_Spec.ts [hunk 1]
```diff
@@ -161,4 +161,37 @@ describe("Autocomplete bug fixes", function () {
       "console.log('hello')",
     );
   });
+
+  it("11. Bug #23641 Verifies if 'children' shows up in autocomplete list", function () {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Autocomplete/Autocomplete_Spec.ts:165]: Not more than 10 specs in a file. Its making people tired if they want to run some spec as they do not know the dependencies. So from next time - please consider creating new spec in such cases.

**Review by Aishwarya-U-R — APPROVED** [review]
Cypress looks good!

**Pull request merged**